eclair-core/src/main/scala/fr/acinq/eclair/wire/internal/channel/version3/ChannelCodecs3.scala

```diff
@@ -0,0 +1,324 @@
+    val claimRemoteDelayedOutputTxCodec: Codec[ClaimRemoteDelayedOutputTx] = (("inputInfo" | inputInfoCodec) :: ("tx" | txCodec)).as[ClaimRemoteDelayedOutputTx]
+    val mainPenaltyTxCodec: Codec[MainPenaltyTx] = (("inputInfo" | inputInfoCodec) :: ("tx" | txCodec)).as[MainPenaltyTx]
+    val htlcPenaltyTxCodec: Codec[HtlcPenaltyTx] = (("inputInfo" | inputInfoCodec) :: ("tx" | txCodec)).as[HtlcPenaltyTx]
+    val claimHtlcDelayedOutputPenaltyTxCodec: Codec[ClaimHtlcDelayedOutputPenaltyTx] = (("inputInfo" | inputInfoCodec) :: ("tx" | txCodec)).as[ClaimHtlcDelayedOutputPenaltyTx]
+    val claimLocalAnchorOutputTxCodec: Codec[ClaimLocalAnchorOutputTx] = (("inputInfo" | inputInfoCodec) :: ("tx" | txCodec)).as[ClaimLocalAnchorOutputTx]
+    val claimRemoteAnchorOutputTxCodec: Codec[ClaimRemoteAnchorOutputTx] = (("inputInfo" | inputInfoCodec) :: ("tx" | txCodec)).as[ClaimRemoteAnchorOutputTx]
+    val closingTxCodec: Codec[ClosingTx] = (("inputInfo" | inputInfoCodec) :: ("tx" | txCodec) :: ("outputIndex" | optional(bool8, outputInfoCodec))).as[ClosingTx]
+
+    val txWithInputInfoCodec: Codec[TransactionWithInputInfo] = discriminated[TransactionWithInputInfo].by(uint16)
+      .typecase(0x01, commitTxCodec)
+      .typecase(0x02, htlcSuccessTxCodec)
+      .typecase(0x03, htlcTimeoutTxCodec)
+      .typecase(0x04, claimHtlcSuccessTxCodec)
+      .typecase(0x05, claimHtlcTimeoutTxCodec)
+      .typecase(0x06, claimP2WPKHOutputTxCodec)
+      .typecase(0x07, claimLocalDelayedOutputTxCodec)
+      .typecase(0x08, mainPenaltyTxCodec)
+      .typecase(0x09, htlcPenaltyTxCodec)
+      .typecase(0x10, closingTxCodec)
+      .typecase(0x11, claimLocalAnchorOutputTxCodec)
+      .typecase(0x12, claimRemoteAnchorOutputTxCodec)
+      .typecase(0x13, claimRemoteDelayedOutputTxCodec)
+      .typecase(0x14, claimHtlcDelayedOutputPenaltyTxCodec)
+      .typecase(0x15, htlcDelayedTxCodec)
+
+    val claimRemoteCommitMainOutputTxCodec: Codec[ClaimRemoteCommitMainOutputTx] = discriminated[ClaimRemoteCommitMainOutputTx].by(uint8)
+      .typecase(0x01, claimP2WPKHOutputTxCodec)
+      .typecase(0x02, claimRemoteDelayedOutputTxCodec)
+
+    val claimAnchorOutputTxCodec: Codec[ClaimAnchorOutputTx] = discriminated[ClaimAnchorOutputTx].by(uint8)
+      .typecase(0x01, claimLocalAnchorOutputTxCodec)
+      .typecase(0x02, claimRemoteAnchorOutputTxCodec)
+
+    val htlcTxCodec: Codec[HtlcTx] = discriminated[HtlcTx].by(uint8)
+      .typecase(0x01, htlcSuccessTxCodec)
+      .typecase(0x02, htlcTimeoutTxCodec)
+
+    val claimHtlcTxCodec: Codec[ClaimHtlcTx] = discriminated[ClaimHtlcTx].by(uint8)
+      .typecase(0x01, claimHtlcSuccessTxCodec)
+      .typecase(0x02, claimHtlcTimeoutTxCodec)
+
+    val htlcTxsAndRemoteSigsCodec: Codec[HtlcTxAndRemoteSig] = (
+      ("txinfo" | htlcTxCodec) ::
+        ("remoteSig" | lengthDelimited(bytes64))).as[HtlcTxAndRemoteSig]
```

Comment:
nit: isn't it an opportunity to remove the `lengthDelimited` around `bytes64` now that we don't have any ambiguity between der and compact sigs?